service/lib/dinstaller/config.rb

```diff
@@ -76,19 +77,32 @@ def data
       return @data if @data
 
       @data = @pure_data.dup || {}
-      pick_product(@data["products"].keys.first) if @data["products"]
+      pick_product(products.keys.first) unless products.empty?
       @data
     end
 
     def pick_product(product)
       data.merge!(data[product])
     end
 
+    # list of available base products for current architecture
+    def products
+      return @products if @products
+      return [] unless @pure_data["products"]
+
+      # cannot use `data` here to avoid endless loop as in data we use
+      # pick_product that select product from products
+      @products = @pure_data["products"].select do |_key, value|
+        value["archs"].nil? ||
+          Yast2::ArchFilter.from_string(value["archs"]).match?
```

Comment:
I am not against the current solution, but I wonder whether it might be better (or even possible) to removed/adjust the elements in a generic way. For instance, you are filtering the products here, so the `#products` method needs to know about that. What if it already gets a filtered representation?

Reply:
My point is that we could do it one level up, taking cmdline and other stuff into consideration. However, I am not sure whether it would work in all cases. Perhaps something that we can explore after the December Prototype.